core/src/main/java/org/apache/calcite/rel/metadata/RelMdUtil.java

```diff
@@ -713,9 +714,11 @@ public static Double getJoinRowCount(RelMetadataQuery mq, Join join,
       // semijoin filter and pass it to getSelectivity
       RexNode semiJoinSelectivity =
           RelMdUtil.makeSemiJoinSelectivityRexNode(mq, join);
-
+      Double selectivity = mq.getSelectivity(join.getLeft(), semiJoinSelectivity);
       return NumberUtil.multiply(
-          mq.getSelectivity(join.getLeft(), semiJoinSelectivity),
+          join.getJoinType() == JoinRelType.SEMI
+              ? selectivity
+              : 1D - selectivity, // ANTI join
```

Comment:
This would fail with NPE in case selectivity is null

Reply:
Fixed.